cprover-cpp-api/src/api.cpp

```diff
@@ -0,0 +1,54 @@
+// Author: Fotis Koutoulakis for Diffblue Ltd.
+
+#include "api.h"
+
+#include <util/cmdline.h>
+#include <util/config.h>
+#include <util/message.h>
+#include <util/options.h>
+
+#include <goto-programs/goto_model.h>
+#include <goto-programs/initialize_goto_model.h>
+
+#include <ansi-c/ansi_c_language.h>
+#include <langapi/mode.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+extern configt config;
+
+api_sessiont::api_sessiont()
+  : message_handler(
+      util_make_unique<null_message_handlert>(null_message_handlert{})),
+    options(util_make_unique<optionst>(optionst{}))
+{
+  // Needed to initialise the language options correctly
+  cmdlinet cmdline;
```

Comment:
Is this the CBMC commandline? Doesn't that make a weird dependency?

Reply:
Yes - hence why it's handled internally :)